heat/core/linalg.py

```diff
@@ -496,40 +617,70 @@ def c_block_setter(b_proc, a_proc, a_data, b_data):
+                        b_proc=b.comm.rank,
+                        a_data=a_lp_data[pr - 1],
+                        b_data=b._DNDarray__array,
+                    )
 
                     # check if there is a remainder on b in the previous node
-                    if pr - 1 in a_rem_locs1:  # this loop is intended to get the remainders of b since it is the one being passed
-                        a_rem[:, pr - 1] = a_lp_data[pr - 1][:, -1]  # takes care of the remainders in b as well as dim0 of a
-
-                    if b_rem_locs1.nelement() != 0:  # this loop is to take care of the remainders in dim1 of B
+                    if (
+                        pr - 1 in a_rem_locs1
+                    ):  # this loop is intended to get the remainders of b since it is the one being passed
+                        a_rem[:, pr - 1] = a_lp_data[pr - 1][
+                            :, -1
+                        ]  # takes care of the remainders in b as well as dim0 of a
+
+                    if (
+                        b_rem_locs1.nelement() != 0
+                    ):  # this loop is to take care of the remainders in dim1 of B
                         if r_loc is not None:
                             st = index_map[pr - 1, 0, 1, 0].item()
                             sp = index_map[pr - 1, 0, 1, 1].item()
-                            c._DNDarray__array[:, r_loc.item()] += (a_lp_data[pr - 1] @ r[st:sp, None]).flatten()
+                            c._DNDarray__array[:, r_loc.item()] += (
+                                a_lp_data[pr - 1] @ r[st:sp, None]
+                            ).flatten()
 
                     del a_lp_data[pr - 1]
 
                 # need to wait if its the last loop, also need to collect the remainders
                 if pr == b.comm.size - 1:
                     req[pr].wait()
-                    c_block_setter(a_proc=pr, b_proc=a.comm.rank, a_data=a_lp_data[pr], b_data=b._DNDarray__array)
+                    c_block_setter(
+                        a_proc=pr,
+                        b_proc=a.comm.rank,
+                        a_data=a_lp_data[pr],
+                        b_data=b._DNDarray__array,
+                    )
                     # check if there is a remainder on b on the last node (there shouldnt be)
                     if pr in a_rem_locs1:
-                        a_rem[:, pr] = a_lp_data[pr][:, -1]  # this is to save the data from B required by the remainders from dim1 of A
+                        a_rem[:, pr] = a_lp_data[pr][
+                            :, -1
+                        ]  # this is to save the data from B required by the remainders from dim1 of A
 
-                    if b_rem_locs1.nelement() != 0:  # this loop is to take care of the remainders in the 0th dimension of A
+                    if (
+                        b_rem_locs1.nelement() != 0
+                    ):  # this loop is to take care of the remainders in the 0th dimension of A
```

Comment:
Comment is wrong. If is no loop

This also seems wrong, comment should be above